tests/integration-tests/tests/networking/test_security_groups.py

```diff
@@ -106,19 +122,30 @@ def test_overwrite_sg(region, scheduler, custom_security_group, pcluster_config_
         _assert_security_group_rules(
             ec2_client,
             mount_target_security_group_id,
-            [mount_target_security_group_id, custom_security_group_id],
+            [mount_target_security_group_id, custom_security_group_ids[0]],
         )
-
     if scheduler == "slurm":
         logging.info("Checking SSH connection between cluster nodes before cluster update")
         _check_connections_between_head_node_and_compute_nodes(cluster)
+
+        # Update the Security Groups of HN to check if the HN instance is replaced.
+        updated_config_file = pcluster_config_reader(
+            config_file="pcluster.config.update_sg.yaml",
+            fsx_name=fsx_name,
+            efs_name=efs_name,
+            vpc_security_group_id=custom_security_group_ids,
+            additional_security_groups=additional_security_groups,
+        )
+        cluster.update(str(updated_config_file), force_update="true")
+        logging.info("Checking if the HeadNode is replaced after cluster update")
```

Comment:
the message should be: `Verifying the HeadNode is not replaced after cluster update` otherwise it seems we're expecting it has to be replaced.